### PR TITLE
[FLINK-2955]  Add operators description in Table API page

### PR DESCRIPTION
**Pull request by ChengXiangLi**
On the Table API page, there is no formal introduction of current supported operations, it should be nice to have it.

### CODE REVIEW

File: docs/libs/table.md [hunk 1]
```diff
@@ -123,13 +123,205 @@ DataSet<WC> result = tableEnv.toDataSet(filtered, WC.class);
 When using Java, the embedded DSL for specifying expressions cannot be used. Only String expressions
 are supported. They support exactly the same feature set as the expression DSL.
 
-## Expression Syntax
+## Table API Operators
+Table API provide a domain-spcific language to execute language-integrated query on structured data in Scala and Java.
```
Comment (fhueske) [inline docs/libs/table.md]: query -> queries

File: docs/libs/table.md [hunk 2]
```diff
@@ -123,13 +123,205 @@ DataSet<WC> result = tableEnv.toDataSet(filtered, WC.class);
...
+This section gives a brief overview of all available operators. You can find more details of operators in the [Javadoc](http://flink.apache.org/docs/latest/api/java/org/apache/flink/api/table/Table.html).
```
Comment (fhueske) [inline docs/libs/table.md]: We shouldn't put a hard link to the latest JavaDocs here but link to the correct version.

File: docs/libs/table.md [hunk 3]
```diff
@@ -123,13 +123,205 @@ DataSet<WC> result = tableEnv.toDataSet(filtered, WC.class);
...
+Table in = tableEnv.fromDataSet(ds, "a, b, c");
+Table result = in.select("a, b, c as d");
+{% endhighlight %}
+      </td>
+    </tr>
+
+    <tr>
+      <td><strong>As</strong></td>
+      <td>
+        <p>Rename the fields name.</p>
```
Comment (fhueske) [inline docs/libs/table.md]: "Rename the fields name" is redundant. How about "Rename fields"?

File: docs/libs/table.md [hunk 4]
```diff
@@ -123,13 +123,205 @@ DataSet<WC> result = tableEnv.toDataSet(filtered, WC.class);
...
+  </thead>
+
+  <tbody>
+    <tr>
+      <td><strong>Select</strong></td>
+      <td>
+        <p>Similar to a SQL SELECT statement. Perform a select operation.</p>
+{% highlight java %}
+Table in = tableEnv.fromDataSet(ds, "a, b, c");
+Table result = in.select("a, b, c as d");
```
Comment (fhueske) [inline docs/libs/table.md]: Leave out `a` or `b` to demonstrate the projection?

File: docs/libs/table.md [hunk 5]
```diff
@@ -123,13 +123,205 @@ DataSet<WC> result = tableEnv.toDataSet(filtered, WC.class);
...
+    </tr>
+
+    <tr>
+      <td><strong>GroupBy</strong></td>
+      <td>
+        <p>Similar to a SQL GROUPBY clause. Group the elements on the grouping keys, with a following aggregation</p>
+        <p>operator to aggregate on per-group basis.</p>
+{% highlight java %}
+Table in = tableEnv.fromDataSet(ds, "a, b, c");
+Table result = in.groupby("a").select("a, b.sum");
```
Comment (fhueske) [inline docs/libs/table.md]: can you assign an attribute name to `b.sum`?

File: docs/libs/table.md [hunk 6]
```diff
@@ -123,13 +123,205 @@ DataSet<WC> result = tableEnv.toDataSet(filtered, WC.class);
...
+      </td>
+    </tr>
 
-A `Table` supports to following operations: `select`, `where`, `groupBy`, `join` (Plus `filter` as
-an alias for `where`.). These are also documented in the [Javadoc](http://flink.apache.org/docs/latest/api/java/org/apache/flink/api/table/Table.html) 
-of Table.
+    <tr>
+      <td><strong>Join</strong></td>
+      <td>
+        <p>Similar to a SQL JOIN clause. Join two tables, the fields of two joined tables must not overlap.</p>
```
Comment (fhueske) [inline docs/libs/table.md]: I would rephrase "the fields of two joined tables must not overlap" to "both tables must have distinct field names". It would also be good to mention that the join condition is mandatory and specified with `where()`.

File: docs/libs/table.md [hunk 7]
```diff
@@ -123,13 +123,205 @@ DataSet<WC> result = tableEnv.toDataSet(filtered, WC.class);
...
+Table result = left.join(right).where("a = d").select("a, b, e");
+{% endhighlight %}
+      </td>
+    </tr>
 
-Some of these expect an expression. These can either be specified using an embedded Scala DSL or
+    <tr>
+      <td><strong>Union</strong></td>
+      <td>
+        <p>Similar to a SQL UNION ALL clause. Union two tables, the fields of two unioned tables must overlap.</p>
```
Comment (fhueske) [inline docs/libs/table.md]: Rephrase "the fields of two unioned tables must overlap" to "Both tables must have identical schema (field names and types)."

**Comment by fhueske** [comment]
Thanks for extending the documentation of the Table API. I added a few comments to the Java section. Most apply to the Scala section as well.

Thanks, Fabian

File: docs/libs/table.md (hunk 2 above)
Comment (ChengXiangLi) [inline docs/libs/table.md]: Where should i link to? The javdoc link of 1.0 version does not exist yet, should i link to 1.0 here anyway, or link to 0.9 here, and let release manager update to 1.0 while release 1.0?

Comment (fhueske) [inline docs/libs/table.md]: If you use, "{{ site.baseurl }}/api/java/org/apache/flink/api/table/Table.html", the correct version will be injected when the markdown file is compiled to html.

**Comment by aljoscha** [comment]
@fhueske is this good to merge now?

**Comment by aljoscha** [comment]
I'm merging

**Comment by aljoscha** [comment]
Thanks for adding this. :+1:

**Comment by aljoscha** [comment]
Github seems to not want to close this. @ChengXiangLi Could you please do it?
